realm/realm-library/src/main/cpp/jni_impl/android_logger.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstring>
+
+#include "android_logger.hpp"
+
+using namespace realm;
+using namespace realm::jni_util;
+using namespace realm::jni_impl;
+
+std::shared_ptr<AndroidLogger> AndroidLogger::shared()
+{
+    // Private constructor, make_shared is not available.
+    static std::shared_ptr<AndroidLogger> android_logger(new AndroidLogger());
+    return android_logger;
+}
+
+void AndroidLogger::log(Log::Level level, const char* tag, jthrowable, const char* stacktrace,
+        const char* message) {
+    android_LogPriority android_log_priority;
+    switch (level) {
+        case Log::Level::trace:
+            android_log_priority = ANDROID_LOG_VERBOSE;
+            break;
+        case Log::Level::debug:
+            android_log_priority = ANDROID_LOG_DEBUG;
+            break;
+        case Log::Level::info:
+            android_log_priority = ANDROID_LOG_INFO;
+            break;
+        case Log::Level::warn:
+            android_log_priority = ANDROID_LOG_WARN;
+            break;
+        case Log::Level::error:
+            android_log_priority = ANDROID_LOG_ERROR;
+            break;
+        case Log::Level::fatal:
+            android_log_priority = ANDROID_LOG_FATAL;
+            break;
+        default:
+            android_log_priority = ANDROID_LOG_VERBOSE; // Cannot get here. Remove the warning only.
+            break;
+    }
+    if (stacktrace) {
```

Comment:
Currently no. I would like to only have the message there. And format the message as stacktrace when the throwable existing?